packages/fxa-settings/src/components/TermsPrivacyAgreement/index.tsx

```diff
@@ -0,0 +1,76 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React from 'react';
+import { FtlMsg } from 'fxa-react/lib/utils';
+import { Link } from '@reach/router';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+
+export type TermsPrivacyAgreementProps = {
+  isPocketClient?: boolean;
+};
+
+const TermsPrivacyAgreement = ({
+  isPocketClient = false,
+}: TermsPrivacyAgreementProps) => {
+  return (
+    <div className="text-grey-500 my-5 text-xs">
+      {isPocketClient ? (
+        <>
+          <FtlMsg id="signin-tos-list-intro">
+            <p className="text-xs">By proceeding, you agree to:</p>
+          </FtlMsg>
+          <ul>
+            <FtlMsg id="signin-tos-list-pocket">
```

Comment:
Ooooh, I actually did not know about `elems={{`... It's quite possible that is a required pattern.

@bcolsson Do you have any insight on this - should we be using `elems={{` in the code any time there are embedded elements (like a <Link> or <Br />) in the string?

If that's the case, I'll file a ticket to audit and fix this in fxa-settings in general, because I'm pretty sure I've missed this in previous PRs.

Reply:
@LZoog - thanks for pointing this out. It looks like this may have slipped through the cracks as I'm not familiar with how fluent/react works and had been relying on your team for the non-linguistic side of things. 

Going a bit through the [documentation](https://github.com/projectfluent/fluent.js/wiki/Localized#markup), it does indeed seem like `elems={{` would be required here. Unfortunately it also looks like we'll need to revise the string to have two separate elements (one for each link). 

Going through existing strings, this would would be a good example to refer to:
[packages/fxa-payments-server/src/components/PaymentConsentCheckbox/en.ftl](https://github.com/mozilla/fxa/blob/e0037d0cc9a165e326f1bdaaa8adb59a67e77f57/packages/fxa-payments-server/src/components/PaymentConsentCheckbox/en.ftl):
`payment-confirm-with-legal-links-static = I authorize { -brand-name-mozilla }, maker of { -brand-name-firefox } products, to charge my payment method for the amount shown, according to <termsOfServiceLink>Terms of Service</termsOfServiceLink> and <privacyNoticeLink>Privacy Notice</privacyNoticeLink>, until I cancel my subscription.`

[packages/fxa-payments-server/src/components/PaymentConsentCheckbox/index.tsx](https://github.com/mozilla/fxa/blob/c0642c045141d233f4c4ae8d795d6dbb7d78c6d9/packages/fxa-payments-server/src/components/PaymentConsentCheckbox/index.tsx):
```tsx
    <Localized
      id="payment-confirm-with-legal-links-static"
      elems={{
        termsOfServiceLink: <a href={termsOfService}>Terms of Service</a>,
        privacyNoticeLink: <a href={privacyNotice}>Privacy Notice</a>,
      }}
    >
      <Checkbox name="confirm" data-testid="confirm" onClick={onClick} required>
        I authorize Mozilla, maker of Firefox products, to charge my payment
        method for the amount shown, according to{' '}
        <a href={termsOfService}>Terms of Service</a> and{' '}
        <a href={privacyNotice}>Privacy Notice</a>, until I cancel my
        subscription.
      </Checkbox>
    </Localized>
```

Reply:
Hmm @vpomerleau I think that doesn't work since that fallback text should use the real elements, since it's displayed when something goes wrong with Fluent or the bundle etc.

Can we do it like this? This is untested in the branch so feel free to shoot me down but referencing other `elems={{` in our codebase, can we change this:
```
default-tos-and-privacy = By proceeding, you agree to the <linkExternal>Terms of Service</linkExternal> and <linkExternal>Privacy Notice</linkExternal>.
```

To this, assuming we need a new ID (not sure when the previous strings landed in the l10n repo):
```
default-tos-and-privacy-2 = By proceeding, you agree to the <termsOfServiceLink>Terms of Service</termsOfServiceLink> and <privacyNoticeLink>Privacy Notice</privacyNoticeLink>.
```

Then we have:
```
<FtlMsg id="default-tos-and-privacy-2" elems={{
  termsOfServiceLink: <Link className="link-grey" to="/legal/terms">
    Terms of Service
  </Link>,
  privacyNoticeLink: <Link className="link-grey" to="/legal/privacy">
    Privacy Notice
  </Link>}}
}}>
  <p className="text-xs">
    By proceeding, you agree to the{' '}
    <Link className="link-grey" to="/legal/terms">
      Terms of Service
    </Link>{' '}
    and{' '}
    <Link className="link-grey" to="/legal/privacy">
      Privacy Notice
    </Link>
    .
  </p>
</FtlMsg>
```

If that works, maybe we can DRY it up like this?

```
const termsOfServiceLink = <Link className="link-grey" to="/legal/terms">
  Terms of Service
</Link>;
const privacyNoticeLink = <Link className="link-grey" to="/legal/privacy">
  Privacy Notice
</Link>;

<FtlMsg id="default-tos-and-privacy-2" elems={{
  termsOfServiceLink,
  privacyNoticeLink
}}>
  <p className="text-xs">
    By proceeding, you agree to the{' '}{termsOfServiceLink}
    and{' '}{privacyNoticeLink}.
  </p>
</FtlMsg>
```

If this causes issues with l10n tests, please leave me a note on FXA-6461.

Reply:
Hmmm -- I think it might be worth it be more verbose here for the sake of clarity

Reply:
@LZoog the example I gave is taken straight from the Fluent docs, which gives me the impression that the Localized element looks for a `<confirm>` element to replace with the real element provided in `elems`.

With your example above, wouldn't the underlying `Localized` be expecting a <termsOfServiceLink> element?

Reply:
> Hmmm -- I think it might be worth it be more verbose here for the sake of clarity

You mean the pre-DRY solution? I feel like that can also leave the possibility of updating one element, without updating the matching element :/

> @LZoog the example I gave is taken straight from the Fluent docs, which gives me the impression that the Localized element looks for a <confirm> element to replace with the real element provided in elems.

Oh yeah, you're right. Now I'm extra confused, since we don't seem to do that anywhere in our codebase?

Reply:
I tried to do some digging and [here's an example](https://github.com/mozilla/fx-private-relay/blob/dd935c11b893e96606aa31d0716c8223e45fcf93/frontend/src/pages/faq.page.tsx#L144) of how Relay is doing it. They aren't using fallback text and the value is [set to this](https://github.com/mozilla/fx-private-relay/blob/dd935c11b893e96606aa31d0716c8223e45fcf93/frontend/pendingTranslations.ftl#L81), but this plus reading the doc linked in my previous comment makes me believe the assumption of using whatever we want in fallback text (e.g. not using the same props passed into `elems`) is fine.

Though, I see in Pontoon's repo where [it's done as described in the docs here](https://github.com/mozilla/pontoon/blob/b3a8ed3f18797f3238a6d7de955d4bd4175f2366/translate/src/core/editor/components/KeyboardShortcuts.tsx#L26-L33), but also[ not done in every case like seen here](https://github.com/mozilla/pontoon/blob/85047a2f62987a268fcc9ea212d95900e61b7695/translate/src/modules/fluenteditor/components/RichTranslationForm.tsx#L35-L44).

The example I gave above would match what we're already [doing in several places in fxa and payments, like here](https://github.com/mozilla/fxa/blob/f001c461c4ec345a25b8b44f4ba9419fb6d98953/packages/fxa-payments-server/src/components/PaymentLegalBlurb/index.tsx#L14-L29).

Reply:
Also https://github.com/mozilla/fxa/pull/14822/files#diff-fcf2a08c68c4be71b9298c34f994cb9a126286e107fe05067d1e3918fee43e48 where SubPlat does something like:
```ftl
# <authenticationAppsLink> links to a list of security apps
inline-totp-setup-add-security-link = Add a layer of security to your account by requiring security codes from one of <authenticationAppsLink>these authentication apps</authenticationAppsLink>.
```

Reply:
Thanks all for helping dig into this question. I've followed the pattern that has been previously used in our codebase - it does look like it might be fine to keep the function tags in our fallback text even though that's not what is shown as an example in the Fluent documentation.

I'll file a ticket tomorrow to do a follow-up check in fxa-settings for FtlMsgs that might be missing elems.